docs/customization.rst

```diff
@@ -337,40 +294,92 @@ Configuration Options
 
             To use light and dark modes, this theme supports 2 schemes which are specified by a ``scheme`` field.
 
-            1. The ``default`` scheme for light mode
+            .. code-block:: python
+                :name: scheme-color-conf-example
 
-               .. code-block:: python
+                html_theme_options = {
+                    "palette": { "scheme": "default" }
+                }
 
-                   html_theme_options = {
-                        "palette": { "scheme": "default" }
-                   }
+            .. only:: html
 
-            2. The ``slate`` scheme for dark mode.
+                - :test-color-scheme:`default` (scheme for light mode)
+                - :test-color-scheme:`slate` (scheme for dark mode)
 
-               .. code-block:: python
+                Click one of the listed schemes to see how it looks.
 
-                   html_theme_options = {
-                        "palette": { "scheme": "slate" }
-                   }
+            .. only:: not html
 
+                - ``default`` (scheme for light mode)
+                - ``slate`` (scheme for dark mode)
 
         .. themeconf:: primary
 
+            The primary color is used for the header, the sidebar, text links and several other
+            components.
+
+            .. code-block:: python
+                :name: primary-color-conf-example
+
+                html_theme_options = {
+                    "palette": { "primary": "green" }
+                }
+
             Primary color options are
 
-            :red:`red`, :pink:`pink`, :purple:`purple`, :deep-purple:`deep-purple`, :indigo:`indigo`, :blue:`blue`,
-            :light-blue:`light-blue`, :cyan:`cyan`, :teal:`teal`, :green:`green`, :light-green:`light-green`,
-            :lime:`lime`, :yellow:`yellow`, :amber:`amber`, :orange:`orange`, :deep-orange:`deep-orange`,
-            :brown:`brown`, :grey:`grey`, :blue-grey:`blue-grey`, :black:`black`, and :white:`white`.
+            .. only:: html
+
+                .. jinja:: colors
+
+                    {% for color in supported_primary -%}
+                    :test-color-primary:`{{color}}`
+                    {%- if color != 'white' %},{% if color == 'black' %} and{% endif %} {% endif %}
```

Comment:
Rather than checking for specific colors you could use `loop.index` and similar variables:
https://stackoverflow.com/questions/12145434/how-to-output-loop-counter-in-python-jinja-template

Alternatively could just use a bullet list and not worry about the punctuation

Reply:
Or maybe hlist

Reply:
Yeah, I was thinking hlist as well.

Reply:
done